contracts/test/sdk/IpcContract.t.sol

```diff
@@ -85,43 +87,53 @@ contract RecorderIpcExchange is IpcExchange {
     }
 }
 
-contract IpcExchangeTest is Test {
+contract IpcExchangeTest is Test, IntegrationTestBase {
     using CrossMsgHelper for IpcEnvelope;
-
-    function test_IpcExchange() public {
-        address gateway = vm.addr(1);
-
+    address gateway = vm.addr(1);
+    SubnetID subnetA;
+    SubnetID subnetB;
+    CallMsg callMsg;
+    IpcEnvelope envelope;
+    RecorderIpcExchange exch;
+
+    function setUp() public override {
         address[] memory pathA = new address[](1);
         pathA[0] = vm.addr(2000);
         address[] memory pathB = new address[](1);
         pathB[0] = vm.addr(3000);
 
         // these two subnets are siblings.
-        SubnetID memory subnetA = SubnetID({root: 123, route: pathA});
-        SubnetID memory subnetB = SubnetID({root: 123, route: pathB});
-
-        CallMsg memory callMsg = CallMsg({method: abi.encodePacked(Foo.foo.selector), params: bytes("1234")});
-        IpcEnvelope memory envelope = IpcEnvelope({
-            kind: IpcMsgKind.Transfer,
+        subnetA = SubnetID({root: 123, route: pathA});
+        subnetB = SubnetID({root: 123, route: pathB});
+        callMsg = CallMsg({method: abi.encodePacked(Foo.foo.selector), params: bytes("1234")});
+        envelope = IpcEnvelope({
+            kind: IpcMsgKind.Call,
             from: IPCAddress({subnetId: subnetA, rawAddress: FvmAddressHelper.from(address(100))}),
             to: IPCAddress({subnetId: subnetB, rawAddress: FvmAddressHelper.from(address(200))}),
             value: 1000,
             message: abi.encode(callMsg),
             nonce: 0
         });
 
-        RecorderIpcExchange exch = new RecorderIpcExchange(gateway);
+        exch = new RecorderIpcExchange(gateway);
+    }
+
+    function test_IpcExchangeTestTransferFails() public {
+        envelope.kind = IpcMsgKind.Transfer;
 
         // a transfer; fails because cannot handle.
         vm.expectRevert(IpcHandler.UnsupportedMsgKind.selector);
         vm.prank(gateway);
         exch.handleIpcMessage(envelope);
+    }
 
+    function test_IpcExchangeTestGatewayOnlyFails() public {
         // a call; fails when the caller is not the gateway.
-        envelope.kind = IpcMsgKind.Call;
         vm.expectRevert(IpcHandler.CallerIsNotGateway.selector);
         exch.handleIpcMessage(envelope);
+    }
 
+    function test_IpcExchange() public {
```

Comment:
Would be great if this test gave a short description of the test case, similarly to other tests.